script/release/uploaders/upload.py

```diff
@@ -205,15 +213,17 @@ def purify_extra_data(mm, offset, length):
 
       if header_id in (EXTENDED_TIME_DATA, UNIX_EXTRA_DATA):
         values[0] = STRIPZIP_OPTION_HEADER
-        for i in xrange(2, len(values)):
+        for i in range(2, len(values)):
           values[i] = 0xff
         extra_struct.pack_into(mm, offset, *values)
-      elif header_id != STRIPZIP_OPTION_HEADER:
-        return False
+      if header_id == ZIP64_EXTRA_HEADER:
+        assert header_length == 16
```

Comment:
```suggestion
        assert header_length == zip64_extra_struct.size
```

Reply:
zip_extra_struct contains both 2 8 byte (QQ) and two 2 bytes (header_id, header_length), so the total should be 20 and not 16.

Reply:
Why's the header being parsed twice..?

Reply:
It is parsed three times actually.

1. First time I only get the `header_id` and `header_length` in `extra_header_struct` ("<HH")
2. second time I get also `header_id` + `header_length` and all his body bytes in `extra_struct` ("<HHB*" where B numbers are equal to the second H)
3. third time, only if it is a `ZIP64_EXTRA_HEADER` I read in the `zip64_extra_struct` ("<HHQQ")

HH is alway the header_id and header_length
body meaning changes wrt what is the header_id, and teoretically can be used to store even stenography. It is really dependant on the implementation what data ends in the Extras of each file. What I try to do in the purify_extra_data is handling the most common and, from this commit, also handling zip64, that, even if it's rarely used, we do use it to compress electron releases.

Reply:
I preferred read 3 times in the worst case and render the code more readable instead of playing with back and forth offsets.